crates/codegen/runtime/cargo/crate/src/extensions/semver/mod.rs

```diff
@@ -0,0 +1,3 @@
+pub fn infer_language_version(_input: &str) -> Vec<semver::Version> {
```

Comment:
Suggestion: returning an iterator, in case the user wants to collect into something like `HashSet<>` for quick membership lookups:

```rs
pub fn infer_language_version(_input: &str) -> impl Iterator<Item = semver::Version> {
```